File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque
n=int(sys.stdin.readline())
dq=deque()
for i in range(n):
    cmd=list(sys.stdin.readline().split())
    if cmd[0]=='push':
        dq.append(int(cmd[1]))
    elif cmd[0]=='pop':
        print(-1 if len(dq)==0 else dq.popleft())
    elif cmd[0]=='front':
        print(-1 if len(dq)==0 else dq[0])
    elif cmd[0]=='back':
        print(-1 if len(dq)==0 else dq[-1])
    elif cmd[0]=='size':
        print(len(dq))
    elif cmd[0]=='empty':
        print(1 if len(dq)==0 else 0)

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <deque>
#include <iostream>
#include <string>
using namespace std;

int main(void) {
    int n, num;
    string cmd;
    deque<int> dq;
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> cmd;
        if (cmd == "push") {
            cin >> num;
            dq.push_back(num);
        } else if (cmd == "front") {
            cout << (dq.size() == 0 ? -1 : dq.front()) << "\n";
        } else if (cmd == "back") {
            cout << (dq.size() == 0 ? -1 : dq.back()) << "\n";
        } else if (cmd == "size") {
            cout << dq.size() << "\n";
        } else if (cmd == "empty") {
            cout << dq.empty() << "\n";
        } else if (cmd == "pop") {
            if (dq.size() == 0)
                cout << -1 << "\n";
            else {
                cout << dq.front() << "\n";
                dq.pop_front();
            }
        }
    }
    return 0;
}